lib/ramble/ramble/util/hashing.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022-2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import hashlib
+import spack.util.spack_json as sjson
+
+
+def hash_file(file_path):
+    file_hash = None
+    with open(file_path, 'rb') as f:
+        bytes = f.read()
+        file_hash = hashlib.sha256(bytes).hexdigest()
+    return file_hash
+
+
+def hash_string(string):
+    return hashlib.sha256(string.encode('UTF-8')).hexdigest()
+
+
+def hash_json(in_json):
```

Comment:
Does `dump` impose a strict ordering to the data? It would be neat if this was resilient to order changes 

Also, dump means things with different spacing etc should be fine, right?

Reply:
It orders it based on the YAML order for ramble.yaml. Since it uses a spack_json dict, it's ordered based on creation order.

But we could change it to be ordered by name or something instead.

Reply:
I added some explicit sorting to help make this more resilient to order changes.